### PR TITLE
test(client): fix test path subdirectories

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
This PR addresses the need to incorporate subdirectory paths for mocha tests that were previously missed during the node16 module resolution updates.

Tests in dds/sequence and dds/matrix have additional tests restore. Other changes do not pickup additional tests, but are good for a consistent pattern.

### CODE REVIEW

File: packages/dds/sequence/package.json [hunk 1]
```diff
@@ -109,8 +109,8 @@
 		"test:memory": "mocha --config src/test/memory/.mocharc.cjs",
 		"test:memory-profiling:report": "mocha --config src/test/memory/.mocharc.cjs",
 		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
```
Comment (jason-ha) [inline packages/dds/sequence/package.json:111]: path qualifications are also missing for these.

Comment (jason-ha) [reply packages/dds/sequence/package.json:111]: Er. No. These don't require paths here, but either they should be here or the spec in the mocha config should be updated to ESM.

File: packages/dds/sequence/src/test/fuzz/fuzzUtils.ts [hunk 2]
```diff
@@ -406,7 +407,7 @@ export const defaultFuzzOptions: Partial<DDSFuzzSuiteOptions> = {
 		clientAddProbability: 0.1,
 	},
 	defaultTestCount: 100,
-	saveFailures: { directory: path.join(__dirname, "../../../src/test/fuzz/results") },
+	saveFailures: { directory: path.join(_dirname, "../../../src/test/fuzz/results") },
```
Comment (jason-ha) [inline packages/dds/sequence/src/test/fuzz/fuzzUtils.ts:410]: To use the _dirname from parent, the relative path here will need to be one ../ less.

**Review by jason-ha — CHANGES_REQUESTED** [review]
test:memory changes to use ESM not required in this PR.

**Comment by sonalideshpandemsft** [comment]
> test:memory changes to use ESM not required in this PR.

should this get removed for CJS too?

File: packages/dds/sequence/package.json [hunk 3]
```diff
@@ -109,8 +109,8 @@
...
-		"test:mocha:cjs": "mocha --ignore \"dist/test/memory/**/*\" --recursive \"dist/test/*.spec.*js\" --exit",
-		"test:mocha:esm": "mocha --ignore \"lib/test/memory/**/*\" --recursive \"lib/test/*.spec.*js\" --exit",
+		"test:mocha:cjs": "mocha --ignore \"dist/test/memory/**/*\" --recursive \"dist/test/**/*.spec.*js\" --exit",
+		"test:mocha:esm": "mocha --ignore \"lib/test/memory/**/*\" --recursive \"lib/test/**/*.spec.*js\" --exit",
```
Comment (anthony-murphy) [inline packages/dds/sequence/package.json:113]: should this be updated everywhere, as i see this pattern used in other packages too. i think all packages should support sub-dirs regardless of if they use them today or not

Comment (jason-ha) [reply packages/dds/sequence/package.json:113]: agree. I haven't noticed that change in other PRs I've reviewed, but then 4 reviewers missed it in this prior one.

File: packages/dds/sequence/package.json [hunk 4]
```diff
@@ -110,7 +110,7 @@
 		"test:memory-profiling:report": "mocha --config src/test/memory/.mocharc.cjs",
 		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
 		"test:mocha:cjs": "mocha --ignore \"dist/test/memory/**/*\" --recursive \"dist/test/**/*.spec.*js\" --exit",
-		"test:mocha:esm": "mocha --ignore \"lib/test/memory/**/*\" --recursive \"lib/test/**/*.spec.*js\" --exit",
+		"test:mocha:esm": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit",
```
Comment (jason-ha) [inline packages/dds/sequence/package.json:113]: Why is memory no longer being ignored?

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: packages/dds/sequence/package.json (hunk 4 above)
Comment (sonalideshpandemsft) [reply packages/dds/sequence/package.json]: oops. reverted this back. why is memory ignored?

File: packages/dds/matrix/package.json [hunk 5]
```diff
@@ -88,8 +88,8 @@
 		"test:memory": "mocha --config src/test/memory/.mocharc.cjs",
 		"test:memory-profiling:report": "mocha --config src/test/memory/.mocharc.cjs",
 		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
-		"test:mocha:cjs": "mocha  --recursive \"dist/test/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
-		"test:mocha:esm": "mocha  --recursive \"lib/test/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha:cjs": "mocha  --recursive \"dist/test/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha:esm": "mocha  --recursive \"lib/test/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline packages/dds/matrix/package.json:92]: This is a required test restoration. Regressed in #19790

**Review by jason-ha — APPROVED** [review]

File: packages/dds/sequence/package.json (hunk 3 above)
Comment (jason-ha) [reply packages/dds/sequence/package.json:113]: I did a survey of where tests may have been dropped. The only other case appeared to be in dds/matrix, which is now also fixed here.

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**